src/wallet/bdk_wallet/wallet.rs

```diff
@@ -23,5 +52,462 @@ pub struct Wallet {
+                xpub.full_derivation_path().expect("").to_string()
+            );
+
+            // Fingerprint calculated from xpub must match with its master
+            let master_fingerprint = wallet
+                .meta_store
+                .master_key
+                .fingerprint(&wallet.secp)
+                .to_string();
+            assert_eq!(xpub.master_fingerprint().to_string(), master_fingerprint);
+        }
+
+        // KeychainTxoutIndex must contain two keychains
+        assert_eq!(wallet.indexed_graph.index.keychains().len(), 2);
+    }
+
+    #[test]
+    fn test_load_wallet() {
+        let (bitcoind, initiated_wallet) = init_bitcoind_and_wallet();
+
+        // Get data_dir from initiated wallet
+        let data_dir = initiated_wallet.data_dir.clone();
+
+        // Get path for meta_store
+        let wallet_file_path = data_dir.join("test_wallet.cbor");
+
+        // Make rpc_config from bitcoind
+        let url = bitcoind.rpc_url().split_at(7).1.to_string();
+        let auth = Auth::CookieFile(bitcoind.params.cookie_file.clone());
+        let network = bitcoind.client.get_blockchain_info().unwrap().chain;
+        let rpc_config = RPCConfig {
+            url,
+            auth,
+            network,
+            wallet_name: "test_wallet".to_string(),
+        };
+
+        // Load the wallet
+        let mut loaded_wallet =
+            Wallet::load(&wallet_file_path, &rpc_config).expect("failed to load wallet");
+
+        // Check that the wallet contains signers for `External` and `Internal` keychains
+        let signers = loaded_wallet.signers;
+        assert_eq!(signers.len(), 2);
+        assert!(signers.contains_key(&Keychain::External));
+        assert!(signers.contains_key(&Keychain::Internal));
+
+        // Create a local chain changeset, apply it to the wallet's local chain,
+        // reload the wallet, and verify that it contains the updates
+        let chain_changeset = LocalChangeSet::from([(1, Some(BlockHash::all_zeros()))]);
+
+        loaded_wallet
+            .chain
+            .apply_changeset(&chain_changeset)
+            .expect("failed to apply changesets to local chain");
+
+        let wallet_changeset = ChangeSet::from(chain_changeset);
+
+        // Open BDK store
+        let mut bdk_store: BDKStore<ChangeSet> =
+            BDKStore::open(&MAGIC_BYTES, data_dir.join("bdk_store.dat")).expect("must not fail");
+
+        // Set the file pointer of BDK store file to end position
+        let _ = bdk_store
+            .aggregate_changesets()
+            .expect("failed to set file pointer to end");
+
+        // Save the changeset to BDK store
+        bdk_store
+            .append_changeset(&wallet_changeset)
+            .expect("failed to append changeset");
+
+        // Reload the wallet from the file system and check if the updates are present
+        let reloaded_wallet =
+            Wallet::load(&wallet_file_path, &rpc_config).expect("failed to load wallet");
```

Comment:
Add a TODO here along the line "Relead wallet after doing some wallet operations". We will add it later when we have wallet operation APIs done, like get a new address, or make a transaction.

We wanna assert that load is working after wallet has operations.

Reply:
Adding a `get_keychain_address` API where it returns a new address for a keychain could be simple and we would need it anyway. Adding that here will allow us to test the load-after-ops scenario.